controllers/data/deployment.go

```diff
@@ -42,7 +57,10 @@ func (r *SnoopyDataEndpointReconciler) deploymentForDataEndpoint(dataEndpoint *d
 			},
 		},
 	}
-	// Set dataEndpoint instance as the owner and controller
-	controllerutil.SetControllerReference(dataEndpoint, deploy, r.Scheme)
+	// Set dataEndpoint instance as the owner and controller.
+	if err := controllerutil.SetControllerReference(dataEndpoint, deploy, r.Scheme); err != nil {
+		return dataEndpoint.DeepCopy()
+	}
```

Comment:
This is a good catch. My question is why is it returning dataEndpoint.DeepCopy() on error? Shouldn't it handle the error or return it? In that case the function signature should change as well and another issue should take care of this.